src/main/java/com/aws/greengrass/certificatemanager/CertificateManager.java

```diff
@@ -132,61 +186,52 @@ public String getCaPassPhrase() {
-                    cisShadowMonitor.addToMonitor(certificateGenerator);
-
-                    certificateGenerator.generateCertificate(connectivityInfoProvider::getCachedHostAddresses,
-                            "initialization of server cert subscription");
-                    return certificateGenerator;
-                } catch (KeyStoreException e) {
-                    logger.atError().setCause(e).log("unable to subscribe to certificate update");
-                    throw new RuntimeException(e);
-                } catch (RuntimeException | NoSuchAlgorithmException | InvalidKeyException | IOException e) {
-                    throw new RuntimeException(new CsrProcessingException(csr, e));
-                }
-            });
-        } catch (RuntimeException e) {
-            if (e.getCause() != null) {
-                Throwable cause = e.getCause();
-                if (cause instanceof KeyStoreException) {
-                    throw (KeyStoreException) cause;
-                }
-                if (cause instanceof CsrProcessingException) {
-                    throw (CsrProcessingException) cause;
-                }
-            }
-            throw e;
+            PKCS10CertificationRequest pkcs10CertificationRequest =
+                    CertificateHelper.getPKCS10CertificationRequestFromPem(csr);
+            JcaPKCS10CertificationRequest jcaRequest =
+                    new JcaPKCS10CertificationRequest(pkcs10CertificationRequest);
+
+            String subject = "temp"; // This is a temporary change, so don't bother properly extracting CN
+            subscribeToServerCertificateUpdatesNoCSR(subject, jcaRequest.getPublicKey(), cb);
+        } catch (CertificateGenerationException | IOException | NoSuchAlgorithmException | InvalidKeyException
+                | RuntimeException e) {
+            throw new CsrProcessingException(csr, e);
         }
     }
 
+    private void subscribeToServerCertificateUpdatesNoCSR(@NonNull String serviceName,
+                                                         @NonNull PublicKey publicKey,
+                                                         @NonNull Consumer<X509Certificate> cb)
+            throws CertificateGenerationException {
+        CertificateGenerator certificateGenerator =
+                new ServerCertificateGenerator(CertificateHelper.getIssuer(serviceName), publicKey, cb,
+                        certificateStore, certificatesConfig, clock);
+
+        // Add certificate generator to monitors first in order to avoid missing events
+        // that happen while the initial certificate is being generated.
+        certExpiryMonitor.addToMonitor(certificateGenerator);
+        cisShadowMonitor.addToMonitor(certificateGenerator);
+
+        certificateGenerator.generateCertificate(connectivityInfoProvider::getCachedHostAddresses,
+                "initialization of server cert subscription");
+
+        serverCertSubscriptions.compute(serviceName, (k, v) -> {
```

Comment:
We didn't have this restriction before, right? I'd be a little worried to keep this. I'm not sure why a component would have more than one subscriber desired, but maybe.

Reply:
I thought about this as well. I don't intend to keep it this way, but this was the easiest way to achieve this without introducing a second map for tracking subscriptions. In the meantime I've added a TODO